pageserver/src/tenant/timeline.rs

```diff
@@ -2819,14 +2902,339 @@ impl Timeline {
             }
         }
     }
+
+    /// Download a layer file from remote storage and insert it into the layer map.
+    ///
+    /// It's safe to call this function for the same layer concurrently.
+    ///
+    /// On success, returns the layer's size.
+    /// Don't use the success value for any acounting purposes, though, since
+    /// every concurrent caller gets that value (=> you would double-account).
+    ///
+    /// If there is a download error, the function retries indefinitely, using an exponential back-off.
+    /// If you need a timeout, simply stop polling: we're cancellation-safe because the download happens in a spearate task_mgr task.
```

Comment:
If you just stop polling, the download continues in the background right? Because it's spawned to a separate task,

```suggestion
    /// If you need a timeout, simply stop polling: we're cancellation-safe because the download happens in a seeparate task_mgr task.
```